src/http.ts

```diff
@@ -3,26 +3,28 @@ import pick from 'lodash.pick';
 import { gzip } from 'zlib';
 import { promisify } from 'util';
 
-import { ErrorCodes, GenericErrorTypes, DEFAULT_ERROR_MESSAGES, MAX_RETRY_ATTEMPTS } from './constants';
+import { DEFAULT_ERROR_MESSAGES, ErrorCodes, GenericErrorTypes, MAX_RETRY_ATTEMPTS } from './constants';
 
 import { BundleFiles, SupportedFiles } from './interfaces/files.interface';
 import { AnalysisResult, ReportResult } from './interfaces/analysis-result.interface';
 import { FailedResponse, makeRequest, Payload } from './needle';
 import {
-  AnalysisOptions,
   AnalysisContext,
+  AnalysisOptions,
   ReportOptions,
   ScmReportOptions,
 } from './interfaces/analysis-options.interface';
-import { getURL } from './utils/httpUtils';
+import { generateJsonApiError, getURL, isJsonApiErrors } from './utils/httpUtils';
 
 type ResultSuccess<T> = { type: 'success'; value: T };
-type ResultError<E> = {
+
+export type ResultError<E> = {
   type: 'error';
   error: {
-    statusCode: E;
+    statusCode: E | number;
     statusText: string;
     apiName: string;
+    detail?: string | undefined;
```

Comment:
`q`: Can we avoid the new `detail` field? Unless we want to show them separately somehow, `statusText` can also include the "friendly" error message currently, so we can do the same with the `jsonApiError` detail message - i.e. `error.title` + `error.detail` (+ link) under `statusText`.

Reply:
I'm not sure as the detail is how we differentiate which errors should be shown to the user Vs hardcoded user messaging [here](https://github.com/snyk/cli/blob/f88d296daaf3b09d236f34af5008542522454f66/src/lib/plugins/sast/index.ts#L167). The statusText is only used in the [debugging output](https://github.com/snyk/cli/blob/f88d296daaf3b09d236f34af5008542522454f66/src/lib/plugins/sast/index.ts#L125) AFAICT.

Reply:
Ah crap we have that custom handling in the CLI too, right. 🙏